File: Projects/HelloObjective-C/HelloObjective-C/Arithmetic.m
//
//  Arithmetic.m
//  HelloObjective-C
//
//  Created by Nimesh Neema on 22/01/25.
//

#import "Arithmetic.h"

// Arithmetic class implementation
@implementation Arithmetic

- (int)sumOfNumber:(int)firstNumber
  withSecondNumber:(int)secondNumber
    andThirdNumber:(int)thirdNumber {
    return firstNumber + secondNumber + thirdNumber;
}

- (int)productOfNumber:(int)firstNumber
      withSecondNumber:(int)secondNumber
        andThirdNumber:(int)thirdNumber {
    return firstNumber * secondNumber * thirdNumber;
}

@end

File: Projects/HelloSwift/HelloSwift/Arithmetic.swift
//
//  Arithmetic.swift
//  HelloSwift
//
//  Created by Nimesh Neema on 22/01/25.
//

import Foundation

    // Arithmetic class
class Arithmetic {

        // Method to calculate the sum of three numbers
        // Signature: (Int, Int, Int) -> Int
    func sum(of firstNumber: Int, withSecondNumber secondNumber: Int, andThirdNumber thirdNumber: Int) -> Int {
        return firstNumber + secondNumber + thirdNumber
    }

        // Method to calculate the product of three numbers
        // Signature: (Int, Int, Int) -> Int
    func product(of firstNumber: Int, withSecondNumber secondNumber: Int, andThirdNumber thirdNumber: Int) -> Int {
        return firstNumber * secondNumber * thirdNumber
    }

        // Signature: (Int, Int, Int) -> Int
    func sum(_ first: Int, secondNumber second: Int, third: Int) -> Int {
        return first + second + third
    }
    
        // Name: sum(first:second:third:)
        // Signature: (Int, Int, Int) -> Int
    func sum(first: Int, second: Int, third:Int) -> Int {
        return first + second + third
    }
}
